automates/text_reading/src/main/scala/org/clulab/aske/automates/OdinActions.scala

```diff
@@ -490,23 +524,149 @@ class OdinActions(val taxonomy: Taxonomy, expansionHandler: Option[ExpansionHand
+          toReturn.append(assembledMention)
+
+        } else toReturn.append(gv._2.head)
+      }
+    }
+    toReturn.distinct ++ noVar
+  }
+
+  def processCommands(mentions: Seq[Mention], state: State = new State()): Seq[Mention] = {
+    val groupBySent = mentions.groupBy(_.sentence)
+    val toReturn = new ArrayBuffer[Mention]()
+    for (g <- groupBySent) {
+      val (commands, other) = g._2.partition(_.label == "CommandSequence")
+      for (o <- other) toReturn.append(o)
+      val otherGroupedByLabel = other.groupBy(_.label)
+      val commandStartSameAsSentStart = commands.filter(m => m.tokenInterval.start == 0)
+      for (c <- commandStartSameAsSentStart) {
+        val curArgs = c.arguments
+        val newArgs = mutable.Map[String, Seq[Mention]]()
+        // for every other mention group
+        for (og <- otherGroupedByLabel.filter(m => (m._1 == "Filename" || m._1 == "Repository" || m._1 == "CommandLineParamValuePair" || m._1 == "CommLineParameter"))) {
+          // there could be multiple mentions that we will want to become command args
+          val newMentionArgValues = new ArrayBuffer[Mention]()
+          for (o <- og._2) {
+            // if the other mention is completely subsumed by command mention, add the other mention to command args
+            if (o.tokenInterval.intersect(c.tokenInterval).length == o.tokenInterval.length) {
+              newMentionArgValues.append(o)
+            }
+          }
+
+          def firstCharLower(string: String): String = {
+            string.head.toLower + string.tail
+          }
+          if (newMentionArgValues.nonEmpty) {
+            newArgs(firstCharLower(og._1)) = newMentionArgValues.distinct
+          }
+
+        }
+
+        val (newTrigger, args) = (curArgs ++ newArgs.toMap).partition(_._1 == "command")
+        val trigger = newTrigger.values.flatten.head.asInstanceOf[TextBoundMention]
+        toReturn.append(copyWithArgs(c, args).asInstanceOf[RelationMention].toEventMention(trigger))
+      }
+    }
+    toReturn
+  }
+
+  def locationsAreNotVariablesOrModels(mentions: Seq[Mention], state: State = new State()): Seq[Mention] = {
+    // makes sure there are no mentions where variables are actually most likely locations
+    // for now only applied to descriptions, but may want to extend to other mentions
+    val (locations, nonLocations) = mentions.partition(_.label == "Location")
+    val (phrases, nonPhrases) = nonLocations.partition(_.label == "Phrase")
+    val noLocations = new ArrayBuffer[Mention]()
+
+    def isInLocations(sent: Int, tokInt: Interval, locations: Seq[Mention]): Boolean = {
+      locations.exists(loc => loc.sentence == sent & loc.tokenInterval == tokInt)
+    }
+
+    for (m <- nonPhrases) {
+
+      val args = m.arguments.values.flatten
+      m match {
+        case m: TextBoundMention => {
+
+          // todo: should I filter out tbm that look like identifiers but are locations?
```

Comment:
I tried it a couple of weeks ago and I have doubts---some identifiers are labeled as locations by the NER and we lose identifiers... I am inclined to not filter that way for now, but keep in mind that it might be reasonable thing to filter in future.